### ISSUE TITLE
ddpg with gumbel softmax reparameterization for discrete action space

### ISSUE THREAD

**Issue by Usaywook** [Usaywook/rl_baseline#2]
- gumbel softmax의 temperature는 낮을 수록 One hot의 형태로 나오지만 variance가 크기 때문에 argmax를 했을 때 decision 결과가 많이 변동될 수 있다. 따라서 1.0 정도로 fix하는 것이 좋았다.  
 - discrete action space에서 actor의 출력은 logit을 의미하기 때문에 continuous action 에 대한 physical constraint를 고려하기 위한 squash function으로 사용되던 tanh 를 제거한다.  
 - physical control과 같은 관성이 있는 action이 아니기 때문에  normal noise를 더하여 exploration을 한다.
 - standard normal noise는 exploration이 잘 안된다. 
 - annealed Gaussian noise를 더하여 start std : 10.0 에서 0 으로 linear decay 로 scheduling하여 개선한다.
